sdk/storage/AzureStorageBlob/Source/Operation/BlockOperations.swift

```diff
@@ -77,7 +77,7 @@ internal class BlockOperation: TransferOperation {
                         return
                     }
                     let blobProperties = BlobProperties(from: responseHeaders)
-                    let bytesTransferred = blobProperties.contentLength ?? 0
+                    let bytesTransferred = (blobProperties.contentLength ?? 1) - 1
```

Comment:
Off-by-one error here would result in the assertion always being triggered.

Reply:
Do you understand why this is the case? I would think if we are summing up the content-length of all the blocks, it should exactly equal the total bytes.

Reply:
A difference between the range being inclusive or exclusive.

Reply:
After correcting the off-by-one errors, the sum of bytes does equal the total, which is how I knew it was cumulative off-by-one errors rather than something weird from the service.

Reply:
This one still puzzles me, because I would think that regardless of whether we are calculating the range wrong, the content-length returned from the server should always 100% match the exact number of bytes it's returning. Now maybe we are requesting one too many bytes when we split things into chunks (because of Range inclusive vs exclusive), and that's causing each result to provide one too many bytes back?